src/robotlibcore.py

```diff
@@ -174,6 +175,34 @@ def __join_defaults_with_types(self, method, types):
                 types[name] = type(value)
         return types
 
+    def get_keyword_source(self, keyword_name):
+        method = self.__get_keyword(keyword_name)
+        path = self.__get_keyword_path(method)
+        line_number = self.__get_keyword_line(method)
+        if not path and not line_number:
+            return None
+        if not path:
+            return ':%s' % line_number
+        if not line_number:
+            return path
+        return '%s:%s' % (path, line_number)
+
+    def __get_keyword_line(self, method):
+        try:
+            source, line_number = inspect.getsourcelines(method)
+        except (OSError, IOError):
```

Comment:
I think this can also fail with `TypeError`.

Reply:
Could be, added.